uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/DAO/WorkDay.php

```diff
@@ -0,0 +1,329 @@
+   *
+   * @var float
+   */
+  public $break;
+  /**
+   * The proportion of a days leave that will be deducted if this day is taken as leave.
+   *
+   * @var int unsigned
+   */
+  public $leave_days;
+  /**
+   *
+   * @var int unsigned
+   */
+  public $number_of_hours;
+  /**
+   * The Work Week this Day belongs to
+   *
+   * @var int unsigned
+   */
+  public $week_id;
+  /**
+   * class constructor
+   *
+   * @access public
+   * @return civicrm_hrleaveandabsences_work_day
+   */
+  function __construct()
+  {
+    $this->__table = 'civicrm_hrleaveandabsences_work_day';
+    parent::__construct();
+  }
+  /**
+   * return foreign keys and entity references
+   *
+   * @static
+   * @access public
+   * @return array of CRM_Core_Reference_Interface
+   */
+  static function getReferenceColumns()
+  {
+    if (!self::$_links) {
+      self::$_links = static ::createReferenceColumns(__CLASS__);
+      self::$_links[] = new CRM_Core_Reference_Basic(self::getTableName() , 'week_id', 'civicrm_hrleaveandabsences_work_week', 'id');
+    }
+    return self::$_links;
+  }
+  /**
+   * returns all the column names of this table
+   *
+   * @access public
+   * @return array
+   */
+  static function &fields()
+  {
+    if (!(self::$_fields)) {
+      self::$_fields = array(
+        'id' => array(
+          'name' => 'id',
+          'type' => CRM_Utils_Type::T_INT,
+          'required' => true,
+        ) ,
+        'day_of_the_week' => array(
+          'name' => 'day_of_the_week',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Day Of The Week') ,
+          'required' => true,
+        ) ,
+        'type' => array(
+          'name' => 'type',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Type') ,
+          'required' => true,
+        ) ,
+        'time_from' => array(
+          'name' => 'time_from',
+          'type' => CRM_Utils_Type::T_TIME,
+          'title' => ts('Time From') ,
+        ) ,
+        'time_to' => array(
+          'name' => 'time_to',
+          'type' => CRM_Utils_Type::T_TIME,
+          'title' => ts('Time To') ,
+        ) ,
+        'break' => array(
+          'name' => 'break',
+          'type' => CRM_Utils_Type::T_MONEY,
+          'title' => ts('Break') ,
+          'precision' => array(
+            20,
+            2
+          ) ,
+        ) ,
+        'leave_days' => array(
+          'name' => 'leave_days',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Leave Days') ,
+        ) ,
+        'number_of_hours' => array(
+          'name' => 'number_of_hours',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Number Of Hours') ,
+        ) ,
+        'week_id' => array(
+          'name' => 'week_id',
+          'type' => CRM_Utils_Type::T_INT,
+          'required' => true,
+          'FKClassName' => 'CRM_HRLeaveAndAbsences_DAO_WorkWeek',
+        ) ,
+      );
+    }
+    return self::$_fields;
+  }
```

Comment:
These files are automatically generated with GenCode.php. Any change made to them will be overwritten